### ISSUE TITLE
Odo url should support both devfile and s2i components

### ISSUE THREAD

**Issue by GeekArthur** [openshift/odo#2761]
/kind bug

## What versions of software are you using?

**Operating System:**
```
MacOS Catalina
```

**Output of `odo version`:**
```
odo v1.1.1 (5ee77dd23)

Server: https://jing-okd-master.rtp.raleigh.ibm.com:8443
OpenShift: v3.11.0+3a34b96-217
Kubernetes: v1.11.0+d4cacc0
```

## Bug description
Here is the output from our OpenShift CI integration tests
```
[odo]  •  Checking if the specified component type is supported devfile component type  ...
[odo]  ✗  Checking if the specified component type is supported devfile component type [48716ns]
[odo] 
[odo] Please run 'odo catalog list components' for a list of supported devfile component types
[odo]  •  Validating component  ...
[odo] 
 ✓  Validating component [162ms]
[odo] 
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args [odo url create --port 8080]
[odo]  ✗  cluster host must be provided in order to create ingress
```
From the output we can clearly see the `odo url create --port 8080` doesn't work if creating s2i component with experimental mode is enabled. By design when experimental mode is enabled we should support both devfile and s2i components but devfile component has higher priority, currently both `odo create` and `odo push` already implemented the design, `odo url` should implement the design as well.

**Comment by kadel** [comment]
TODO for me: Add more information 
/triage needs-information

**Comment by GeekArthur** [comment]
@kadel The bug is `odo url create --port 8080` fails with experimental mode enabled given user uses `odo create` to create a s2i component beforehand. This issue should be fixed by PR #2762.
FYI @yangcao77

**Comment by mik-dass** [comment]
@GeekArthur So the steps to reproduce this are
```
odo preference set experimental false -f
odo create nodejs
odo preference set experimental true -f
odo url create example --port 8080
```
Are these the right steps?

**Comment by GeekArthur** [comment]
@mik-dass Yup, that's right steps, or even more simple
```
odo preference set experimental true -f
odo create python
odo url create example --port 8080  //fails here 
```
Because currently devfile component doesn't support python, you can create s2i python component directly with experimental mode enabled to save one step to switch experimental mode

**Comment by mik-dass** [comment]
> Because currently devfile component doesn't support python, you can create s2i python component directly with experimental mode enabled

But currently while creating a python component with experimental mode enabled, it panics.

```
mrinaldas@localhost nodejs]$ odo create pyhton
 ✗  Checking if the specified component type is supported devfile component type [118942ns]

Please run 'odo catalog list components' for a list of supported devfile component types
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd8 pc=0x169378c]

goroutine 1 [running]:
github.com/openshift/odo/pkg/config.(*LocalConfigInfo).ConfigFileExists(...)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/config/config.go:324
github.com/openshift/odo/pkg/odo/cli/component.(*CreateOptions).Complete(0xc000286780, 0xc00054a8a0, 0x6, 0xc000704000, 0xc000716cc0, 0x1, 0x1, 0x2, 0x1c0f99d)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/odo/cli/component/create.go:404 +0x68c
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x1f60c80, 0xc000286780, 0xc000704000, 0xc000716cc0, 0x1, 0x1)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:29 +0xb8
github.com/openshift/odo/pkg/odo/cli/component.NewCmdCreate.func1(0xc000704000, 0xc000716cc0, 0x1, 0x1)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/odo/cli/component/create.go:705 +0x5e
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).execute(0xc000704000, 0xc000716ca0, 0x1, 0x1, 0xc000704000, 0xc000716ca0)
        /home/mrinaldas/go/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:702 +0x289
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00035c000, 0x1d41150, 0xc000468c00, 0x0)
        /home/mrinaldas/go/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:783 +0x2ca
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /home/mrinaldas/go/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:736
main.main()
        /home/mrinaldas/go/src/github.com/openshift/odo/cmd/odo/odo.go:75 +0x463
```

**Comment by mik-dass** [comment]
> @kadel The bug is odo url create --port 8080 fails with experimental mode enabled given user uses odo create to create a s2i component beforehand. This issue should be fixed by PR #2762.

#2762 seems to have fixed this issue. Can you please check and close this issue if already resolved.

**Comment by GeekArthur** [comment]
@mik-dass I am pretty sure this use case `creating a python component with experimental mode enabled` works before, #2762 is merged, seems like it fixes this issue but cause the regression of failing to create s2i component with experimental mode enabled

**Comment by GeekArthur** [comment]
We should verify this issue until the regression has been fixed.

**Comment by mik-dass** [comment]
using https://github.com/openshift/odo/pull/2839, I tried the above steps again and it seems the issue has been fixed by #2762 . Can you please verify it?

```
[mrinaldas@localhost nodejs]$ odo create python
 ✗  Checking if the specified component type is supported devfile component type [101668ns]

Please run 'odo catalog list components' for a list of supported devfile component types
Warning: python is not fully supported by odo, and it is not guaranteed to work
 ✓  Validating component [819ms]

Please use `odo push` command to create the component with source deployed
[mrinaldas@localhost nodejs]$ odo url create example
 ✓  URL example created for component: python-nodejs-fxjz

To create URL on the OpenShift Cluster, please use `odo push`
```

**Comment by girishramnani** [comment]
as mentioned by @mik-dass, this issue got resolved by a PR
